OpenRA.Game/TraitDictionary.cs

```diff
@@ -227,10 +228,12 @@ public IEnumerable<Actor> Actors(Func<T, bool> predicate)
 
 				for (var i = 0; i < actors.Count; i++)
 				{
-					if (actors[i] == last || !predicate(traits[i]))
-						continue;
-					yield return actors[i];
 					last = actors[i];
```

Comment:
Setting this here and then comparing it with the same value in the `if` guarantees that we will never yield anything!

Reply:
Sorry, I didn't notice this line because it isn't marked as changed.

Reply:
Sorry. Made original change in another branch.